File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
    return (number * (number + 1) * (2 * number + 1) / 6);
}

unsigned int square_of_sum(unsigned int number) {
    float numf = (float)number;

    return (unsigned int)powf(numf * (numf + 1.0F) / 2.0F, 2);
}

unsigned int difference_of_squares(unsigned int number) {
    return square_of_sum(number) - sum_of_squares(number);
}

File: ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/difference-of-squares
# Difference of Squares exercise
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    (@number * (@number + 1) / 2)**2
  end

  def sum_of_squares
    @number * (@number + 1) * (2 * @number + 1) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
